_posts/2024-06-13-installing-maven-style-embulk-plugins.md

```diff
@@ -0,0 +1,197 @@
+
+```
+plugins {
+    id "org.embulk.runset" version "0.2.0"  // Just apply this Gradle plugin.
+}
+
+repositories {
+    mavenCentral()
+}
+
+installEmbulkRunSet {
+    // Set your Embulk home directory (absolute path) to install the Embulk plugins and "embulk.properties".
+    embulkHome file("/home/user/my-embulk-home")
+
+    // Specify the Maven-style Embulk plugin by the "artifact" directive.
+    artifact "org.embulk:embulk-input-s3:0.6.0"
+
+    // You can specify multiple versions of the same Embulk plugin so that you can choose the version at runtime.
+    // You can also specify an artifact with the split-style notation.
+    artifact group: "org.embulk", name: "embulk-input-s3", version: "0.5.3"
+
+    // Specify this if you need JRuby.
+    // It downloads jruby-complete-9.1.15.0.jar, and set the "jruby" Embulk System Property in "embulk.properties".
+    jruby "org.jruby:jruby-complete:9.1.15.0"
+
+    // Specify this if you need to set some Embulk System Properties manually.
+    // It sets the "key" Embulk System Property to "value" in "embulk.properties".
+    embulkSystemProperty "key", "value"
+}
+```
+
+Then, run `gradle installEmbulkRunSet` (`./gradlew` when you use the Gradle wrapper) to set up.
+
+```
+$ gradlew installEmbulkRunSet
+
+> Configure project :
+Supplied embulkHome "/home/user/my-embulk-home" does not exist, then will be created.
+Setting to copy org.embulk:embulk-input-s3:0.6.0:jar into org/embulk/embulk-input-s3/0.6.0
+Setting to copy com.amazonaws:aws-java-sdk-s3:1.11.466:jar into com/amazonaws/aws-java-sdk-s3/1.11.466
+...
+...
+Setting to copy org.embulk:embulk-input-s3:0.5.3:jar into org/embulk/embulk-input-s3/0.5.3
+...
+...
+Setting to copy org.embulk:embulk-input-s3:0.5.3:pom into org/embulk/embulk-input-s3/0.5.3
+...
+...
+Setting to copy org.jruby:jruby-complete:9.1.15.0:jar into org/jruby/jruby-complete/9.1.15.0
+
+BUILD SUCCESSFUL in 2s
+1 actionable task: 1 executed
+```
+
+The Embulk System Properties file `embulk.properties` is automatically generated in the specified Embulk home, too.
+
+```
+#Generated by the "org.embulk.embulk-runset" Gradle plugin.
+#Thu Jun 13 16:53:31 JST 2024
+key=value
+jruby=file\:///home/user/my-embulk-home/lib/m2/repository/org/jruby/jruby-complete/9.1.15.0/jruby-complete-9.1.15.0.jar
+```
+
+## Run!
+
+In either style of installation, you can run Embulk with the installed Maven-style Embulk plugins.
+
+See the example `s3_with_maven.yaml` below.
+
+```yaml
+in:
+  # The full-style type notation for Maven-style Embulk plugins.
+  type:
+    source: maven
+    group: org.embulk
+    name: s3
+    version: 0.6.0
+  bucket: ...
+  parser:
+    type: csv
+    ...
+out:
+  type: stdout
+```
+
+Then, run Embulk!
+
+```
+$ java -jar embulk-0.11.4.jar -Xembulk_home=/home/user/my-embulk-home run s3_with_maven.yml
+2024-06-13 17:01:55.373 +0900 [INFO] (main): embulk_home is set from command-line: /home/user/my-embulk-home
```

Comment:
Minor: Above example used `embulk-0.11.3`. I thought Using the same Embulk version give a more unified look.

Reply:
Ah, maybe. Thanks! Let me update it.

Reply:
Finally, I concluded that the current revision is okay. It's because:
* The section 1 is about the feature introduced in v0.11.3, then using v0.11.3.
* The section 2 is not about Embulk's built-in feature, then using just the latest one.